_posts/2019-07-26-pysam-with-s3.md

```diff
@@ -0,0 +1,188 @@
+```
+[D::init_add_plugin] Loaded "knetfile"[D::init_add_plugin] Loaded "mem"
+[D::init_add_plugin] Loaded "/usr/local/libexec/htslib/hfile_s3_write.so"
+[D::init_add_plugin] Loaded "/usr/local/libexec/htslib/hfile_s3.so"
+[D::init_add_plugin] Loaded "/usr/local/libexec/htslib/hfile_libcurl.so"
+[D::init_add_plugin] Loaded "/usr/local/libexec/htslib/hfile_gcs.so"
+*   Trying 147.75.201.93...
+* TCP_NODELAY set
+* Connected to play.min.io (147.75.201.93) port 9000 (#0)
+* ALPN, offering http/1.1
+* successfully set certificate verify locations:
+*   CAfile: /etc/ssl/certs/ca-certificates.crt
+  CApath: /etc/ssl/certs
+* SSL connection using unknown / TLS_AES_128_GCM_SHA256
+* ALPN, server accepted to use http/1.1
+* Server certificate:
+*  subject: CN=play.minio.io
+*  start date: Jul  2 08:22:01 2019 GMT
+*  expire date: Sep 30 08:22:01 2019 GMT
+*  subjectAltName: host "play.min.io" matched cert's "play.min.io"
+*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
+*  SSL certificate verify ok.
+> GET /testfiles/NA18537.vcf.gz HTTP/1.1
+Host: play.min.io:9000
+User-Agent: htslib/1.9-258-ga428aa2 libcurl/7.58.0
+Accept: */*
+Authorization: AWS4-HMAC-SHA256 Credential=Q3AM3UQ867SPQQA43P2F/20190710/us-east-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=c4dcdc1b3d49a33b231151f760fb54f07606f0c0016c002847cd646d94d0c34b
+x-amz-date: 20190710T204159Z
+x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+
+< HTTP/1.1 200 OK
+< Accept-Ranges: bytes
+< Content-Length: 3185107
+< Content-Security-Policy: block-all-mixed-content
+< Content-Type: application/octet-stream
+< ETag: "c876ca31f911ade7751eb23b4827253c"
+< Last-Modified: Thu, 04 Jul 2019 17:02:02 GMT
+< Server: MinIO/DEVELOPMENT.2019-07-06T17-09-12Z
+< Vary: Origin
+< X-Amz-Bucket-Region: us-east-1
+< X-Amz-Request-Id: 15B0266CE660B4AD
+< X-Amz-Server-Side-Encryption: AES256
+< X-Xss-Protection: 1; mode=block
+< Date: Wed, 10 Jul 2019 20:41:59 GMT
+< 
+s3://default@play/testfiles/NA18537.vcf.gz:     VCF version 4.1 BGZF-compressed variant calling data
+* stopped the pause stream!
+* Closing connection 0
+```
+
+Pay close attention to the "Host:". If your host is not being read correctly, play around with the host_base in ./s3cfg file and the file path you pass in with the htsfile command
+
+#### 4) Rebuild PySAM from source
+If you already have PySAM installed, you need to uninstall it. We want to rebuild PySAM from source to make it use our custom htslib instead of the release version.
+
+To install PySAM from source, refer to this link: https://github.com/pysam-developers/pysam 
+
+##### Clone the repository:
+```
+git clone https://github.com/pysam-developers/pysam.git
+```
+
+Change your directory into root of PySAM. 
+
+##### Install cython
+Since pysam depends on cython, it has to be installed beforehand:
+```
+pip install cython
+```
+
+##### Export environment variables
```

Comment:
I'd add a little text here: "This is the crucial step - it specifies that PySAM should use the htslib version that we built and installed above, rather than the default version that PySAM itself ships with".